### DISCUSSION TITLE
force get_to to parse as string?

### DISCUSSION THREAD

**Discussion by smnrock** [nlohmann/json#3713] [Q&A] [answered]
I've a array stored in json string and array can be either float array or string array. I want to read this array in to std::vector<string> irrespective of whether it's float or string array.

std::vector<string> values;
Jsonstring.get_to(values);

This kind of code works only if it the array is string, but fails for float array.

Any help?

**Comment by falbrechtskirchinger** [comment, marked as answer]
There's no library mechanism to perform the conversion for you. Just use a loop or `std` algorithms as usual.

```cpp
std::vector<std::string> values;
values.reserve(Jsonstring.size());
std::transform(Jsonstring.begin(), Jsonstring.end(),
               std::back_inserter(values), [](const json &j) {
                   if (j.is_number())
                       return std::to_string(j.get<double>());
                   else
                       return j.get<std::string>();
               });
```
